app/scaffold-controllers/navigationController.js

```diff
@@ -52,12 +57,32 @@ function(
             }
 
             var navigationController = new NavigationController(
-                tab,
+                id,
+                url,
                 layout,
                 navigationView,
                 navigationHeaderController,
                 drawerIconEnabled
             );
+
+            Astro.events.on('welcome:shown', function() {
```

Comment:
This code looks a bit repetitive. Can we extract it into a function and call it twice?